service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -402,7 +401,7 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
                         attributes.addAll(deserializedAttributes);
                         // This enforces the limit on the count of attributes returned from the traversal, instead of
                         // number of vertices traversed. To do the latter will require traversing the graph twice.
-                        checkTraversalLimitOrFail(attributes);
+                        checkTraversalLimitOrFail(entity, attributes);
```

Comment:
Rather than going about it this way, consider using the `sideEffect()` intermediate in place of subject. `sideEffect()` allows you to inspect the current state of the graph traverser (similar to peek() in Java 8 streams). For example:

`.sideEffect( t -> { if ( t.loops() > limit ) doSomething(); })`

This would limit based on the number of loops the traverser has gone through. There might be ways to limit based on results as well. Not sure, but we should explore other options.